core/src/main/java/com/graphhopper/routing/util/parsers/AbstractAccessParser.java

```diff
@@ -78,16 +78,18 @@ public ConditionalTagInspector getConditionalTagInspector() {
     /**
      * Updates the given edge flags based on node tags
      */
-    protected void handleNodeTags(IntsRef edgeFlags, Map<String, Object> nodeTags) {
-        if (!nodeTags.isEmpty()) {
+    protected void handleNodeTags(IntsRef edgeFlags, List<Map<String, Object>> nodeTags) {
+        if (nodeTags.isEmpty()) return; // allMatch returns 'true' by default
+        boolean isBarrier = nodeTags.stream().allMatch(tags -> {
```

Comment:
I think we should do this already one level higher in OSMReader#handleEdge#setArtificialWayTags, something like:

```java
        if (!nodeTags.isEmpty() && (boolean) nodeTags.get(0).getOrDefault("gh:split_node", false))
            way.setTag("gh:split_node_edge", true);
        else
            way.removeTag("gh:split_node_edge");
```

This way the downstream tag parsers can recognize edges that were created by splitting a node by looking at the `gh:split_node_edge` tag and can access the tags of the split node by looking at the node tags for either of the two nodes (they are the same anyway).

Reply:
We can do this even earlier: The moment we create the barrier edge we now add the `gh:barrier_edge` tag for the reader way.

We could even use the node tags of the split node as value of the `gh:barrier_edge` tag (instead of simply `true`)? But we can also obtain these tags in the parsers by looking at the node tags of either of the two nodes of these edges, like we currently do, not sure if one would be more clear than the other. We could even do both :)